tests/test_hubspot_automatic_fields_test.py

```diff
@@ -0,0 +1,132 @@
+import tap_tester.connections as connections
+import tap_tester.menagerie   as menagerie
+import tap_tester.runner      as runner
+import re
+
+from base import HubspotBaseTest
+
+class TestHubspotAutomaticFields(HubspotBaseTest):
+    def name(self):
+        return "tap_tester_hubspot_combined_test"
+
+    @staticmethod
+    def get_selected_fields_from_metadata(metadata):
+        selected_fields = set()
+        for field in metadata:
+            is_field_metadata = len(field['breadcrumb']) > 1
+            inclusion_automatic_or_selected = (field['metadata'].get('inclusion') == 'automatic'
+                                               or field['metadata'].get('selected') is True)
+            if is_field_metadata and inclusion_automatic_or_selected:
+                selected_fields.add(field['breadcrumb'][1])
+        return selected_fields
+
+    def get_properties(self):
+        return {
+            'start_date' : '2016-06-02T00:00:00Z',
+        }
+
+    def expected_check_streams(self):
+        return {
+            "subscription_changes",
+            "email_events",
+            "forms",
+            "workflows",
+            "owners",
+            "campaigns",
+            "contact_lists",
+            "contacts",
+            "companies",
+            "deals",
+            "engagements",
+            "deal_pipelines",
+            "contacts_by_company"
+        }
+
+    def expected_streams(self):
+        """
+        All streams are under test with the exception of 'budgets' which
+        has a workaround to skip the invalid assertion. See if block in
+        test_run for details
+        """
+        return self.expected_check_streams().difference({
+            'subscription_changes',
+            'email_events',
+            'companies',
+            'contacts',
+            'deals',
+            'engagements',
+            'contacts_by_company',
+        })
+
+
+    def test_run(self):
+        conn_id = self.ensure_connection()
+
+        # Run in check mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # Verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        found_catalogs = menagerie.get_catalogs(conn_id)
+        self.assertGreater(len(found_catalogs), 0, msg="unable to locate schemas for connection {}".format(conn_id))
+
+        # Select only the expected streams tables
+        expected_streams = self.expected_streams()
+        catalog_entries = [ce for ce in found_catalogs if ce['tap_stream_id'] in expected_streams]
+        self.select_all_streams_and_fields(conn_id, catalog_entries, select_all_fields=False)
```

Comment:
With the exception of this line, the majority of this code above and some of the test code below is identical. Can you move them into the base? You can look at tap-square for examples. In my opinion most of this should be moved to be shared tap-tester code... since all our tests do it.

* run_and_verify_check_mode: https://github.com/singer-io/tap-square/blob/master/tests/base.py#L513
* run_and_verify_sync: https://github.com/singer-io/tap-square/blob/master/tests/base.py#L573